// react-bs-notifier/lib/alert/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactJss = require("react-jss");
var _toetag = require("toetag");
var _icon = _interopRequireDefault(require("./icon"));
var _transitionStyles = _interopRequireDefault(require("../transition-styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const Alert = ({
  type = "info",
  children,
  headline,
  onDismiss,
  dismissTitle = "Dismiss",
  showIcon = true,
  classes: classOverrides
}) => {
  let classes = useStyles();
  classes = {
    ...classes,
    ...classOverrides
  };
  const isDismissable = !!onDismiss;
  const css = `${isDismissable ? classes.dismissable : ""} ${classes[type]} ${classes.alert}`;
  const dismiss = isDismissable ? /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    className: classes.close,
    title: dismissTitle,
    onClick: onDismiss
  }, "\xD7") : null;
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("div", {
    className: css
  }, dismiss, showIcon ? /*#__PURE__*/_react.default.createElement(_icon.default, {
    className: classes.icon,
    type: type
  }) : null, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.msgContainer
  }, headline ? /*#__PURE__*/_react.default.createElement("h4", {
    className: classes.headline
  }, headline) : null, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.body
  }, children))));
};
const useStyles = (0, _reactJss.createUseStyles)({
  ..._transitionStyles.default,
  alert: {
    composes: "alert",
    marginBottom: _toetag.bootstrap.paddingBaseVertical,
    display: "inline-block",
    textAlign: "left"
  },
  info: {
    composes: "alert-info"
  },
  success: {
    composes: "alert-success"
  },
  warning: {
    composes: "alert-warning"
  },
  danger: {
    composes: "alert-danger"
  },
  dismissable: {
    composes: "alert-dismissable"
  },
  close: {
    composes: "close"
  },
  msgContainer: {
    display: "inline-block"
  },
  icon: {
    verticalAlign: "top",
    fontSize: _toetag.bootstrap.fontSizeH4,
    paddingRight: _toetag.bootstrap.paddingLargeHorizontal,
    opacity: 0.2
  },
  headline: {
    margin: 0,
    marginBottom: _toetag.bootstrap.paddingBaseVertical
  },
  body: {
    maxWidth: "40em"
  }
});
var _default = Alert;
exports.default = _default;
