// react-bs-notifier/lib/container.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useStyles = exports.default = exports.PropTypes = exports.EXIT_TIMEOUT = exports.ENTER_TIMEOUT = void 0;
var _react = _interopRequireWildcard(require("react"));
var _propTypes = require("prop-types");
var _reactTransitionGroup = require("react-transition-group");
var _alertTransition = _interopRequireDefault(require("./alert-transition"));
var _reactJss = require("react-jss");
var _toetag = require("toetag");
var _transitionStyles = _interopRequireDefault(require("./transition-styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const ENTER_TIMEOUT = 500;
exports.ENTER_TIMEOUT = ENTER_TIMEOUT;
const EXIT_TIMEOUT = 300;
exports.EXIT_TIMEOUT = EXIT_TIMEOUT;
const AlertContainer = ({
  position = "top-right",
  children
}) => {
  const classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: `${classes.container} ${classes[position]}`
  }, /*#__PURE__*/_react.default.createElement(_reactTransitionGroup.TransitionGroup, null, _react.Children.map(children, child => child ? /*#__PURE__*/_react.default.createElement(_alertTransition.default, {
    key: child.props.id
  }, /*#__PURE__*/(0, _react.cloneElement)(child)) : null)));
};
const useStyles = (0, _reactJss.createUseStyles)({
  container: {
    position: "fixed",
    paddingTop: _toetag.bootstrap.paddingBaseVertical,
    paddingRight: _toetag.bootstrap.paddingBaseHorizontal,
    paddingBottom: _toetag.bootstrap.paddingBaseVertical,
    paddingLeft: _toetag.bootstrap.paddingBaseHorizontal,
    zIndex: _toetag.bootstrap.zindexNavbarFixed + 1
  },
  "top-right": {
    top: 0,
    right: 0,
    textAlign: "right"
  },
  "top-left": {
    top: 0,
    left: 0
  },
  "bottom-right": {
    bottom: 0,
    right: 0,
    textAlign: "right"
  },
  "bottom-left": {
    bottom: 0,
    left: 0
  },
  ..._transitionStyles.default
});
exports.useStyles = useStyles;
const PropTypes = {
  position: (0, _propTypes.oneOf)(["top-right", "top-left", "bottom-right", "bottom-left"])
};
exports.PropTypes = PropTypes;
AlertContainer.propTypes = PropTypes;
var _default = AlertContainer;
exports.default = _default;
